frontend/class-frontend.php

```diff
@@ -1147,6 +1147,8 @@ public function metakeywords() {
 			$keywords = WPSEO_Meta::get_value( 'metakeywords' );
 			if ( $keywords === '' && ( is_object( $post ) && ( ( isset( $this->options[ 'metakey-' . $post->post_type ] ) && $this->options[ 'metakey-' . $post->post_type ] !== '' ) ) ) ) {
 				$keywords = wpseo_replace_vars( $this->options[ 'metakey-' . $post->post_type ], $post );
+			} else {
+				$keywords = wpseo_replace_vars( $keywords, $post );
 			}
```

Comment:
May I suggest the following (same for the other one below):
```php
$keywords = wpseo_replace_vars( WPSEO_Meta::get_value( 'metakeywords' ), $post );
if ( $keywords === '' && ( is_object( $post ) && ( ( isset( $this->options[ 'metakey-' . $post->post_type ] ) && $this->options[ 'metakey-' . $post->post_type ] !== '' ) ) ) ) {
	$keywords = wpseo_replace_vars( $this->options[ 'metakey-' . $post->post_type ], $post );
}
```

The logic being that if the post doesn't contain any keywords a fall-back is used.
The variable replacement might 'fail', i.e. contain only variables which yield no values, those will then be parsed out and an empty string would be returned.
So, the replacements should be done *before* the check whether the fall-back is needed.